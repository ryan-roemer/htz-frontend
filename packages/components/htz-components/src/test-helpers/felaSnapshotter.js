import React from 'react';
import createRenderer from '../components/StyleProvider/createRenderer';
import StyleProvider from '../components/StyleProvider/StyleProvider';
import createFelaSnapshotter from './createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: true, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer);

export default felaSnapshotter;
