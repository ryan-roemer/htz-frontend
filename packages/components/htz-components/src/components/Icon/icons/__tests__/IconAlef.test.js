import React from 'react';
import felaSnapshotter from '../../../../test-helpers/felaSnapshotter';
import IconAlef from '../IconAlef';

describe('<IconLogo />', () => {
  describe('Errors', () => {
    it('throw an error when the value passed to the "size" prop isn\'t a number', () => {
      expect(() => felaSnapshotter(<IconAlef size="2rem" />)).toThrow(
        /^An Icon's "size" prop may only be passed a "number"/
      );
    });
    it('throw an error when the value passed to the "color" prop isn\'t a named color (string)', () => {
      expect(() => felaSnapshotter(<IconAlef color="red" />)).toThrow(
        /is not a named-color/
      );
    });
    it('throw an error when the value passed to the "color" prop isn\'t a named color (array of strings)', () => {
      expect(() =>
        felaSnapshotter(<IconAlef color={[ 'primary', 'nope', ]} />)
      ).toThrow(/is not a named-color/);
    });
    it('throw an error when the value passed to the "fill" prop isn\'t a named color (string)', () => {
      expect(() => felaSnapshotter(<IconAlef fill="red" />)).toThrow(
        /is not a named-color/
      );
    });
    it('throw an error when the value passed to the "fill" prop isn\'t a named color (array of strings)', () => {
      expect(() =>
        felaSnapshotter(<IconAlef fill={[ 'primary', 'nope', ]} />)
      ).toThrow(/is not a named-color/);
    });
  });
  describe('DOM element', () => {
    testCase('render correctly when no props are passed', <IconAlef />);
    testCase(
      'pass attributes defined in the "attrs" prop to the DOM element',
      <IconAlef attrs={{ 'aria-hidden': true, }} />
    );
    testCase('add "id" attribute to DOM element', <IconAlef id="hasId" />);
    testCase(
      'add "onClick" attribute to DOM element',
      <IconAlef id={e => console.log('a')} />
    );
  });

  describe('styles', () => {
    testCase("set icon's color", <IconAlef color={[ 'primary', 'base', ]} />);
    testCase(
      "set icon's color responsively",
      <IconAlef
        color={[
          { until: 'xl', value: [ 'primary', 'base', ], },
          { from: 'xl', value: [ 'bodyText', 'base', ], },
        ]}
      />
    );
    testCase("set icon's fill", <IconAlef fill={[ 'primary', 'base', ]} />);
    testCase(
      "set icon's fill responsively",
      <IconAlef
        fill={[
          { until: 'xl', value: [ 'primary', 'base', ], },
          { from: 'xl', value: [ 'bodyText', 'base', ], },
        ]}
      />
    );
    testCase("set icon's size", <IconAlef size={6} />);
    testCase(
      "set icon's size responsively",
      <IconAlef size={[ { until: 'xl', value: 6, }, { from: 'xl', value: 7, }, ]} />
    );
    testCase(
      'set misc styles on an icon',
      <IconAlef miscStyles={{ position: 'relative', }} />
    );
    testCase(
      'responsively set misc styles on an icon',
      <IconAlef
        miscStyles={{
          position: [
            { until: 'xl', value: 'relative', },
            { from: 'xl', value: 'static', },
          ],
        }}
      />
    );
  });
});

function testCase(name, Component, { only = false, } = {}) {
  const testFunc = only ? it.only : it;
  testFunc(name, () => {
    const { component, styles, } = felaSnapshotter(Component);
    expect(component).toMatchSnapshot();
    expect(styles).toMatchSnapshot();
  });
}
