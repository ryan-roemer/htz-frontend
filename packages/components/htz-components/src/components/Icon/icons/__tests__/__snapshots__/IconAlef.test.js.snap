// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconAlef /> DOM element add "id" attribute to DOM element 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    id="hasId"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M10.5 248h93.7c-8-17.1-11.1-37.1-9.4-60H59.3c-13.2 0-21.1-7-23.7-21.1-2.6-14.1-3-27.4-1.3-40.1 1.1-9.8 3.7-16.9 7.8-21.5 4-4.6 6.3-6.9 6.9-6.9l196.4 144.7c0-.6.3-.9.9-.9.6 0 .6-.3 0-.9-2.9-6.3-5.5-12.9-7.8-19.8s-4-13.5-5.2-19.8c-1.2-8.6-1.6-15.1-1.3-19.4.3-4.3 1.3-10.8 3-19.4-11.5-7.5-22.4-15.2-32.7-23.3-10.3-8-23-17.2-37.9-27.6-1.7-1.1-3.3-3.6-4.7-7.3-1.4-3.7-1.9-7.3-1.3-10.8 1.1-5.2 3.4-9.5 6.9-12.9 3.4-3.4 7.2-5.2 11.2-5.2h56.6c-10.9-28.7-11.2-51.4-.9-68h-79.8c0 2.9-.4 6.2-1.3 9.9s-1 8.5-.4 14.2c0 5.7 1.1 11.5 3.4 17.2 2.3 5.7 2.3 12.9 0 21.5-2.3 6.9-4.4 12-6.5 15.5-2 3.4-4.2 5.7-6.5 6.9-18.4-13.8-39-29.1-61.9-46C56.3 30.2 39.1 17.2 27.7 8H12.2c-.6 0-.9.3-.9.9 4 8 6.9 16.9 8.6 26.7 1.7 9.8 2 19.2.9 28.4-1.2 12-2.3 20.5-3.4 25.4-1.2 4.9-1.7 12.5-1.7 22.8 0 11.5 1.1 22.2 3.4 32.3 2.3 10 3.1 20.5 2.6 31.4-1.2 16.1-3 30.3-5.6 42.6-2.6 12.3-4.7 20.8-6.4 25.4-.7 1.5-.4 3 .8 4.1z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlef /> DOM element add "id" attribute to DOM element 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconAlef /> DOM element add "onClick" attribute to DOM element 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    id={[Function]}
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M10.5 248h93.7c-8-17.1-11.1-37.1-9.4-60H59.3c-13.2 0-21.1-7-23.7-21.1-2.6-14.1-3-27.4-1.3-40.1 1.1-9.8 3.7-16.9 7.8-21.5 4-4.6 6.3-6.9 6.9-6.9l196.4 144.7c0-.6.3-.9.9-.9.6 0 .6-.3 0-.9-2.9-6.3-5.5-12.9-7.8-19.8s-4-13.5-5.2-19.8c-1.2-8.6-1.6-15.1-1.3-19.4.3-4.3 1.3-10.8 3-19.4-11.5-7.5-22.4-15.2-32.7-23.3-10.3-8-23-17.2-37.9-27.6-1.7-1.1-3.3-3.6-4.7-7.3-1.4-3.7-1.9-7.3-1.3-10.8 1.1-5.2 3.4-9.5 6.9-12.9 3.4-3.4 7.2-5.2 11.2-5.2h56.6c-10.9-28.7-11.2-51.4-.9-68h-79.8c0 2.9-.4 6.2-1.3 9.9s-1 8.5-.4 14.2c0 5.7 1.1 11.5 3.4 17.2 2.3 5.7 2.3 12.9 0 21.5-2.3 6.9-4.4 12-6.5 15.5-2 3.4-4.2 5.7-6.5 6.9-18.4-13.8-39-29.1-61.9-46C56.3 30.2 39.1 17.2 27.7 8H12.2c-.6 0-.9.3-.9.9 4 8 6.9 16.9 8.6 26.7 1.7 9.8 2 19.2.9 28.4-1.2 12-2.3 20.5-3.4 25.4-1.2 4.9-1.7 12.5-1.7 22.8 0 11.5 1.1 22.2 3.4 32.3 2.3 10 3.1 20.5 2.6 31.4-1.2 16.1-3 30.3-5.6 42.6-2.6 12.3-4.7 20.8-6.4 25.4-.7 1.5-.4 3 .8 4.1z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlef /> DOM element add "onClick" attribute to DOM element 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconAlef /> DOM element pass attributes defined in the "attrs" prop to the DOM element 1`] = `
<div>
  <svg
    aria-hidden={true}
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M10.5 248h93.7c-8-17.1-11.1-37.1-9.4-60H59.3c-13.2 0-21.1-7-23.7-21.1-2.6-14.1-3-27.4-1.3-40.1 1.1-9.8 3.7-16.9 7.8-21.5 4-4.6 6.3-6.9 6.9-6.9l196.4 144.7c0-.6.3-.9.9-.9.6 0 .6-.3 0-.9-2.9-6.3-5.5-12.9-7.8-19.8s-4-13.5-5.2-19.8c-1.2-8.6-1.6-15.1-1.3-19.4.3-4.3 1.3-10.8 3-19.4-11.5-7.5-22.4-15.2-32.7-23.3-10.3-8-23-17.2-37.9-27.6-1.7-1.1-3.3-3.6-4.7-7.3-1.4-3.7-1.9-7.3-1.3-10.8 1.1-5.2 3.4-9.5 6.9-12.9 3.4-3.4 7.2-5.2 11.2-5.2h56.6c-10.9-28.7-11.2-51.4-.9-68h-79.8c0 2.9-.4 6.2-1.3 9.9s-1 8.5-.4 14.2c0 5.7 1.1 11.5 3.4 17.2 2.3 5.7 2.3 12.9 0 21.5-2.3 6.9-4.4 12-6.5 15.5-2 3.4-4.2 5.7-6.5 6.9-18.4-13.8-39-29.1-61.9-46C56.3 30.2 39.1 17.2 27.7 8H12.2c-.6 0-.9.3-.9.9 4 8 6.9 16.9 8.6 26.7 1.7 9.8 2 19.2.9 28.4-1.2 12-2.3 20.5-3.4 25.4-1.2 4.9-1.7 12.5-1.7 22.8 0 11.5 1.1 22.2 3.4 32.3 2.3 10 3.1 20.5 2.6 31.4-1.2 16.1-3 30.3-5.6 42.6-2.6 12.3-4.7 20.8-6.4 25.4-.7 1.5-.4 3 .8 4.1z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlef /> DOM element pass attributes defined in the "attrs" prop to the DOM element 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconAlef /> DOM element render correctly when no props are passed 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M10.5 248h93.7c-8-17.1-11.1-37.1-9.4-60H59.3c-13.2 0-21.1-7-23.7-21.1-2.6-14.1-3-27.4-1.3-40.1 1.1-9.8 3.7-16.9 7.8-21.5 4-4.6 6.3-6.9 6.9-6.9l196.4 144.7c0-.6.3-.9.9-.9.6 0 .6-.3 0-.9-2.9-6.3-5.5-12.9-7.8-19.8s-4-13.5-5.2-19.8c-1.2-8.6-1.6-15.1-1.3-19.4.3-4.3 1.3-10.8 3-19.4-11.5-7.5-22.4-15.2-32.7-23.3-10.3-8-23-17.2-37.9-27.6-1.7-1.1-3.3-3.6-4.7-7.3-1.4-3.7-1.9-7.3-1.3-10.8 1.1-5.2 3.4-9.5 6.9-12.9 3.4-3.4 7.2-5.2 11.2-5.2h56.6c-10.9-28.7-11.2-51.4-.9-68h-79.8c0 2.9-.4 6.2-1.3 9.9s-1 8.5-.4 14.2c0 5.7 1.1 11.5 3.4 17.2 2.3 5.7 2.3 12.9 0 21.5-2.3 6.9-4.4 12-6.5 15.5-2 3.4-4.2 5.7-6.5 6.9-18.4-13.8-39-29.1-61.9-46C56.3 30.2 39.1 17.2 27.7 8H12.2c-.6 0-.9.3-.9.9 4 8 6.9 16.9 8.6 26.7 1.7 9.8 2 19.2.9 28.4-1.2 12-2.3 20.5-3.4 25.4-1.2 4.9-1.7 12.5-1.7 22.8 0 11.5 1.1 22.2 3.4 32.3 2.3 10 3.1 20.5 2.6 31.4-1.2 16.1-3 30.3-5.6 42.6-2.6 12.3-4.7 20.8-6.4 25.4-.7 1.5-.4 3 .8 4.1z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlef /> DOM element render correctly when no props are passed 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconAlef /> styles responsively set misc styles on an icon 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d htz-e"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M10.5 248h93.7c-8-17.1-11.1-37.1-9.4-60H59.3c-13.2 0-21.1-7-23.7-21.1-2.6-14.1-3-27.4-1.3-40.1 1.1-9.8 3.7-16.9 7.8-21.5 4-4.6 6.3-6.9 6.9-6.9l196.4 144.7c0-.6.3-.9.9-.9.6 0 .6-.3 0-.9-2.9-6.3-5.5-12.9-7.8-19.8s-4-13.5-5.2-19.8c-1.2-8.6-1.6-15.1-1.3-19.4.3-4.3 1.3-10.8 3-19.4-11.5-7.5-22.4-15.2-32.7-23.3-10.3-8-23-17.2-37.9-27.6-1.7-1.1-3.3-3.6-4.7-7.3-1.4-3.7-1.9-7.3-1.3-10.8 1.1-5.2 3.4-9.5 6.9-12.9 3.4-3.4 7.2-5.2 11.2-5.2h56.6c-10.9-28.7-11.2-51.4-.9-68h-79.8c0 2.9-.4 6.2-1.3 9.9s-1 8.5-.4 14.2c0 5.7 1.1 11.5 3.4 17.2 2.3 5.7 2.3 12.9 0 21.5-2.3 6.9-4.4 12-6.5 15.5-2 3.4-4.2 5.7-6.5 6.9-18.4-13.8-39-29.1-61.9-46C56.3 30.2 39.1 17.2 27.7 8H12.2c-.6 0-.9.3-.9.9 4 8 6.9 16.9 8.6 26.7 1.7 9.8 2 19.2.9 28.4-1.2 12-2.3 20.5-3.4 25.4-1.2 4.9-1.7 12.5-1.7 22.8 0 11.5 1.1 22.2 3.4 32.3 2.3 10 3.1 20.5 2.6 31.4-1.2 16.1-3 30.3-5.6 42.6-2.6 12.3-4.7 20.8-6.4 25.4-.7 1.5-.4 3 .8 4.1z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlef /> styles responsively set misc styles on an icon 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}@media (max-width: 79.9375em){.htz-d{position:relative}}@media (min-width: 80em){.htz-e{position:static}}"`;

exports[`<IconAlef /> styles set icon's color 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M10.5 248h93.7c-8-17.1-11.1-37.1-9.4-60H59.3c-13.2 0-21.1-7-23.7-21.1-2.6-14.1-3-27.4-1.3-40.1 1.1-9.8 3.7-16.9 7.8-21.5 4-4.6 6.3-6.9 6.9-6.9l196.4 144.7c0-.6.3-.9.9-.9.6 0 .6-.3 0-.9-2.9-6.3-5.5-12.9-7.8-19.8s-4-13.5-5.2-19.8c-1.2-8.6-1.6-15.1-1.3-19.4.3-4.3 1.3-10.8 3-19.4-11.5-7.5-22.4-15.2-32.7-23.3-10.3-8-23-17.2-37.9-27.6-1.7-1.1-3.3-3.6-4.7-7.3-1.4-3.7-1.9-7.3-1.3-10.8 1.1-5.2 3.4-9.5 6.9-12.9 3.4-3.4 7.2-5.2 11.2-5.2h56.6c-10.9-28.7-11.2-51.4-.9-68h-79.8c0 2.9-.4 6.2-1.3 9.9s-1 8.5-.4 14.2c0 5.7 1.1 11.5 3.4 17.2 2.3 5.7 2.3 12.9 0 21.5-2.3 6.9-4.4 12-6.5 15.5-2 3.4-4.2 5.7-6.5 6.9-18.4-13.8-39-29.1-61.9-46C56.3 30.2 39.1 17.2 27.7 8H12.2c-.6 0-.9.3-.9.9 4 8 6.9 16.9 8.6 26.7 1.7 9.8 2 19.2.9 28.4-1.2 12-2.3 20.5-3.4 25.4-1.2 4.9-1.7 12.5-1.7 22.8 0 11.5 1.1 22.2 3.4 32.3 2.3 10 3.1 20.5 2.6 31.4-1.2 16.1-3 30.3-5.6 42.6-2.6 12.3-4.7 20.8-6.4 25.4-.7 1.5-.4 3 .8 4.1z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlef /> styles set icon's color 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:#0B7EB5}.htz-c{fill:#FFF}"`;

exports[`<IconAlef /> styles set icon's color responsively 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M10.5 248h93.7c-8-17.1-11.1-37.1-9.4-60H59.3c-13.2 0-21.1-7-23.7-21.1-2.6-14.1-3-27.4-1.3-40.1 1.1-9.8 3.7-16.9 7.8-21.5 4-4.6 6.3-6.9 6.9-6.9l196.4 144.7c0-.6.3-.9.9-.9.6 0 .6-.3 0-.9-2.9-6.3-5.5-12.9-7.8-19.8s-4-13.5-5.2-19.8c-1.2-8.6-1.6-15.1-1.3-19.4.3-4.3 1.3-10.8 3-19.4-11.5-7.5-22.4-15.2-32.7-23.3-10.3-8-23-17.2-37.9-27.6-1.7-1.1-3.3-3.6-4.7-7.3-1.4-3.7-1.9-7.3-1.3-10.8 1.1-5.2 3.4-9.5 6.9-12.9 3.4-3.4 7.2-5.2 11.2-5.2h56.6c-10.9-28.7-11.2-51.4-.9-68h-79.8c0 2.9-.4 6.2-1.3 9.9s-1 8.5-.4 14.2c0 5.7 1.1 11.5 3.4 17.2 2.3 5.7 2.3 12.9 0 21.5-2.3 6.9-4.4 12-6.5 15.5-2 3.4-4.2 5.7-6.5 6.9-18.4-13.8-39-29.1-61.9-46C56.3 30.2 39.1 17.2 27.7 8H12.2c-.6 0-.9.3-.9.9 4 8 6.9 16.9 8.6 26.7 1.7 9.8 2 19.2.9 28.4-1.2 12-2.3 20.5-3.4 25.4-1.2 4.9-1.7 12.5-1.7 22.8 0 11.5 1.1 22.2 3.4 32.3 2.3 10 3.1 20.5 2.6 31.4-1.2 16.1-3 30.3-5.6 42.6-2.6 12.3-4.7 20.8-6.4 25.4-.7 1.5-.4 3 .8 4.1z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlef /> styles set icon's color responsively 2`] = `".htz-a{vertical-align:-0.15em}.htz-d{fill:#FFF}@media (max-width: 79.9375em){.htz-b{color:#0B7EB5}}@media (min-width: 80em){.htz-c{color:#2D2D2D}}"`;

exports[`<IconAlef /> styles set icon's fill 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M10.5 248h93.7c-8-17.1-11.1-37.1-9.4-60H59.3c-13.2 0-21.1-7-23.7-21.1-2.6-14.1-3-27.4-1.3-40.1 1.1-9.8 3.7-16.9 7.8-21.5 4-4.6 6.3-6.9 6.9-6.9l196.4 144.7c0-.6.3-.9.9-.9.6 0 .6-.3 0-.9-2.9-6.3-5.5-12.9-7.8-19.8s-4-13.5-5.2-19.8c-1.2-8.6-1.6-15.1-1.3-19.4.3-4.3 1.3-10.8 3-19.4-11.5-7.5-22.4-15.2-32.7-23.3-10.3-8-23-17.2-37.9-27.6-1.7-1.1-3.3-3.6-4.7-7.3-1.4-3.7-1.9-7.3-1.3-10.8 1.1-5.2 3.4-9.5 6.9-12.9 3.4-3.4 7.2-5.2 11.2-5.2h56.6c-10.9-28.7-11.2-51.4-.9-68h-79.8c0 2.9-.4 6.2-1.3 9.9s-1 8.5-.4 14.2c0 5.7 1.1 11.5 3.4 17.2 2.3 5.7 2.3 12.9 0 21.5-2.3 6.9-4.4 12-6.5 15.5-2 3.4-4.2 5.7-6.5 6.9-18.4-13.8-39-29.1-61.9-46C56.3 30.2 39.1 17.2 27.7 8H12.2c-.6 0-.9.3-.9.9 4 8 6.9 16.9 8.6 26.7 1.7 9.8 2 19.2.9 28.4-1.2 12-2.3 20.5-3.4 25.4-1.2 4.9-1.7 12.5-1.7 22.8 0 11.5 1.1 22.2 3.4 32.3 2.3 10 3.1 20.5 2.6 31.4-1.2 16.1-3 30.3-5.6 42.6-2.6 12.3-4.7 20.8-6.4 25.4-.7 1.5-.4 3 .8 4.1z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlef /> styles set icon's fill 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#0B7EB5}"`;

exports[`<IconAlef /> styles set icon's fill responsively 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M10.5 248h93.7c-8-17.1-11.1-37.1-9.4-60H59.3c-13.2 0-21.1-7-23.7-21.1-2.6-14.1-3-27.4-1.3-40.1 1.1-9.8 3.7-16.9 7.8-21.5 4-4.6 6.3-6.9 6.9-6.9l196.4 144.7c0-.6.3-.9.9-.9.6 0 .6-.3 0-.9-2.9-6.3-5.5-12.9-7.8-19.8s-4-13.5-5.2-19.8c-1.2-8.6-1.6-15.1-1.3-19.4.3-4.3 1.3-10.8 3-19.4-11.5-7.5-22.4-15.2-32.7-23.3-10.3-8-23-17.2-37.9-27.6-1.7-1.1-3.3-3.6-4.7-7.3-1.4-3.7-1.9-7.3-1.3-10.8 1.1-5.2 3.4-9.5 6.9-12.9 3.4-3.4 7.2-5.2 11.2-5.2h56.6c-10.9-28.7-11.2-51.4-.9-68h-79.8c0 2.9-.4 6.2-1.3 9.9s-1 8.5-.4 14.2c0 5.7 1.1 11.5 3.4 17.2 2.3 5.7 2.3 12.9 0 21.5-2.3 6.9-4.4 12-6.5 15.5-2 3.4-4.2 5.7-6.5 6.9-18.4-13.8-39-29.1-61.9-46C56.3 30.2 39.1 17.2 27.7 8H12.2c-.6 0-.9.3-.9.9 4 8 6.9 16.9 8.6 26.7 1.7 9.8 2 19.2.9 28.4-1.2 12-2.3 20.5-3.4 25.4-1.2 4.9-1.7 12.5-1.7 22.8 0 11.5 1.1 22.2 3.4 32.3 2.3 10 3.1 20.5 2.6 31.4-1.2 16.1-3 30.3-5.6 42.6-2.6 12.3-4.7 20.8-6.4 25.4-.7 1.5-.4 3 .8 4.1z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlef /> styles set icon's fill responsively 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}@media (max-width: 79.9375em){.htz-c{fill:#0B7EB5}}@media (min-width: 80em){.htz-d{fill:#2D2D2D}}"`;

exports[`<IconAlef /> styles set icon's size 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M10.5 248h93.7c-8-17.1-11.1-37.1-9.4-60H59.3c-13.2 0-21.1-7-23.7-21.1-2.6-14.1-3-27.4-1.3-40.1 1.1-9.8 3.7-16.9 7.8-21.5 4-4.6 6.3-6.9 6.9-6.9l196.4 144.7c0-.6.3-.9.9-.9.6 0 .6-.3 0-.9-2.9-6.3-5.5-12.9-7.8-19.8s-4-13.5-5.2-19.8c-1.2-8.6-1.6-15.1-1.3-19.4.3-4.3 1.3-10.8 3-19.4-11.5-7.5-22.4-15.2-32.7-23.3-10.3-8-23-17.2-37.9-27.6-1.7-1.1-3.3-3.6-4.7-7.3-1.4-3.7-1.9-7.3-1.3-10.8 1.1-5.2 3.4-9.5 6.9-12.9 3.4-3.4 7.2-5.2 11.2-5.2h56.6c-10.9-28.7-11.2-51.4-.9-68h-79.8c0 2.9-.4 6.2-1.3 9.9s-1 8.5-.4 14.2c0 5.7 1.1 11.5 3.4 17.2 2.3 5.7 2.3 12.9 0 21.5-2.3 6.9-4.4 12-6.5 15.5-2 3.4-4.2 5.7-6.5 6.9-18.4-13.8-39-29.1-61.9-46C56.3 30.2 39.1 17.2 27.7 8H12.2c-.6 0-.9.3-.9.9 4 8 6.9 16.9 8.6 26.7 1.7 9.8 2 19.2.9 28.4-1.2 12-2.3 20.5-3.4 25.4-1.2 4.9-1.7 12.5-1.7 22.8 0 11.5 1.1 22.2 3.4 32.3 2.3 10 3.1 20.5 2.6 31.4-1.2 16.1-3 30.3-5.6 42.6-2.6 12.3-4.7 20.8-6.4 25.4-.7 1.5-.4 3 .8 4.1z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlef /> styles set icon's size 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}.htz-d{font-size:6rem}"`;

exports[`<IconAlef /> styles set icon's size responsively 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d htz-e"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M10.5 248h93.7c-8-17.1-11.1-37.1-9.4-60H59.3c-13.2 0-21.1-7-23.7-21.1-2.6-14.1-3-27.4-1.3-40.1 1.1-9.8 3.7-16.9 7.8-21.5 4-4.6 6.3-6.9 6.9-6.9l196.4 144.7c0-.6.3-.9.9-.9.6 0 .6-.3 0-.9-2.9-6.3-5.5-12.9-7.8-19.8s-4-13.5-5.2-19.8c-1.2-8.6-1.6-15.1-1.3-19.4.3-4.3 1.3-10.8 3-19.4-11.5-7.5-22.4-15.2-32.7-23.3-10.3-8-23-17.2-37.9-27.6-1.7-1.1-3.3-3.6-4.7-7.3-1.4-3.7-1.9-7.3-1.3-10.8 1.1-5.2 3.4-9.5 6.9-12.9 3.4-3.4 7.2-5.2 11.2-5.2h56.6c-10.9-28.7-11.2-51.4-.9-68h-79.8c0 2.9-.4 6.2-1.3 9.9s-1 8.5-.4 14.2c0 5.7 1.1 11.5 3.4 17.2 2.3 5.7 2.3 12.9 0 21.5-2.3 6.9-4.4 12-6.5 15.5-2 3.4-4.2 5.7-6.5 6.9-18.4-13.8-39-29.1-61.9-46C56.3 30.2 39.1 17.2 27.7 8H12.2c-.6 0-.9.3-.9.9 4 8 6.9 16.9 8.6 26.7 1.7 9.8 2 19.2.9 28.4-1.2 12-2.3 20.5-3.4 25.4-1.2 4.9-1.7 12.5-1.7 22.8 0 11.5 1.1 22.2 3.4 32.3 2.3 10 3.1 20.5 2.6 31.4-1.2 16.1-3 30.3-5.6 42.6-2.6 12.3-4.7 20.8-6.4 25.4-.7 1.5-.4 3 .8 4.1z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlef /> styles set icon's size responsively 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}@media (max-width: 79.9375em){.htz-d{font-size:6rem}}@media (min-width: 80em){.htz-e{font-size:7rem}}"`;

exports[`<IconAlef /> styles set misc styles on an icon 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M10.5 248h93.7c-8-17.1-11.1-37.1-9.4-60H59.3c-13.2 0-21.1-7-23.7-21.1-2.6-14.1-3-27.4-1.3-40.1 1.1-9.8 3.7-16.9 7.8-21.5 4-4.6 6.3-6.9 6.9-6.9l196.4 144.7c0-.6.3-.9.9-.9.6 0 .6-.3 0-.9-2.9-6.3-5.5-12.9-7.8-19.8s-4-13.5-5.2-19.8c-1.2-8.6-1.6-15.1-1.3-19.4.3-4.3 1.3-10.8 3-19.4-11.5-7.5-22.4-15.2-32.7-23.3-10.3-8-23-17.2-37.9-27.6-1.7-1.1-3.3-3.6-4.7-7.3-1.4-3.7-1.9-7.3-1.3-10.8 1.1-5.2 3.4-9.5 6.9-12.9 3.4-3.4 7.2-5.2 11.2-5.2h56.6c-10.9-28.7-11.2-51.4-.9-68h-79.8c0 2.9-.4 6.2-1.3 9.9s-1 8.5-.4 14.2c0 5.7 1.1 11.5 3.4 17.2 2.3 5.7 2.3 12.9 0 21.5-2.3 6.9-4.4 12-6.5 15.5-2 3.4-4.2 5.7-6.5 6.9-18.4-13.8-39-29.1-61.9-46C56.3 30.2 39.1 17.2 27.7 8H12.2c-.6 0-.9.3-.9.9 4 8 6.9 16.9 8.6 26.7 1.7 9.8 2 19.2.9 28.4-1.2 12-2.3 20.5-3.4 25.4-1.2 4.9-1.7 12.5-1.7 22.8 0 11.5 1.1 22.2 3.4 32.3 2.3 10 3.1 20.5 2.6 31.4-1.2 16.1-3 30.3-5.6 42.6-2.6 12.3-4.7 20.8-6.4 25.4-.7 1.5-.4 3 .8 4.1z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlef /> styles set misc styles on an icon 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}.htz-d{position:relative}"`;
